rfc/0002-subdocapi.md

```diff
@@ -112,171 +89,44 @@ support empty paths. All other forms of path verification is done on the server
-> on the server. This means that if the document does not exist, a `KEY_ENOENT`
-> is returned from the server. The document is never created, even if `create_parents`
-> is true
-
-> **ALTERNATIVE SIGNATURES**: Having a `DocumentFragment` class with all the fields necessary for a mutation opens the option of using the `DocumentFragment` as an input parameter for all mutation methods in place of `docid`, `path`, `value`, `cas` and `ttl`.
-
-### `insert_in(docid, path, value, create_parents, cas, ttl, persist_to, replicate_to)`
-Exactly like `upsert_in`, except the path must not already exist.
-If it does, `PathExists` is thrown.
-
-### `replace_in(docid, path, value, cas, ttl, persist_to, replicate_to)`
-Exactly like `upsert_in`, except the penultimate path must exist. If it
-not exist, `PathNotFound` is thrown. Unlike `insert_in` and `upsert_in`,
-`replace_in` can work on both dictionary and array values
-
-### `extend_in(docid, path, value_or_values, direction, create_parents, cas, ttl, persist_to, replicate_to)`
-Add an element to the back or front of an array.
-
-* `value_or_values`: Can be a single JSON value, or a list of multiple
-  JSON values. The list should be of a special type (MultiValue) so the
-  SDK should not serialize the list itself as JSON.
-* `direction`: `FRONT` or `BACK` depending on whether the item is to be
-  added to the beginning or end of the array
-* `create_parents`: Create the enclosing array and any intermediate elements.
-  If this is not set, then the immediate array must exist.
-
-### `arrayinsert_in(docid, path, value_or_values, cas, ttl, persist_to, replicate_to)`
-Insert an element(s) at a given position in the array. The path should indicate
-the desired position at which to insert the new element(s).
-
-**FIXME:** This should really be `insert_in`, but we are already using that for
-the object-based insert.
+## API Methods
 
-### `addunique_in(docid, path, value, create_parents, cas, ttl, persist_to, replicate_to)`
-Adds a 'unique' element into an array (at which end is not defined or specified).
-Functions exactly like `extend_in`. There are some server-side constraints into
-which values can be used for unique operations - currently it is restricted to
-JSON primitives (so no compound items).
+The Subdocument APIs will be exposed through two separate builders which are available through two new top-level Bucket methods `LookupIn` and `MutateIn`.  Additionally, both `MutateIn` and `LookupIn` when executed will result in a `DocumentFragment` object, which represents the top-level results (Status, CAS, MutationToken) and also contains the result of all subdocument operations which were executed.
 
-* Throws `PathExists` if the unique element already exists
-* Throws `CantInsertValue` if the value is not a JSON primitive.
-* Throws `PathMismatch` if the array already contains non-primitive elements
+### LookupInSet
+This builder exposes the creation of a set of lookup operations to be performed.
```

Comment:
Like you said.  It's basically a vaguely ambiguous indication of what you said.  The reason for the current ambiguity is there is still some discussion on whether the API should be a builder which is then 'applied' against a bucket, or if it should be a builder returned by a bucket method.